src/reverse/Type.cpp

```diff
@@ -90,6 +90,41 @@ std::string Type::GetName() const
     return "";
 }
 
+std::string Type::FunctionDescriptor(RED4ext::CBaseFunction* pFunc) const
+{
+    std::stringstream ret;
+
+    // name2 seems to be a cleaner representation of the name
+    // for example, name would be "DisableFootstepAudio;Bool", and name2 is just"DisableFootstepAudio"
+    std::string funcName2 = pFunc->name2.ToString();
+
+    ret << funcName2 << "(";
+
+    RED4ext::CName name;
+    for (auto i = 0u; i < pFunc->params.size; ++i)
+    {
+        auto* param = pFunc->params[i];
+        ret << param->name.ToString() << ": ";
```

Comment:
This should filter "out" params and add them to the return types

Reply:
How do I identify a param is an "out" param? Presumably it's bit on flags. If you can tell me a function that has out params and which one, I can attempt to figure it out.

Hmmm, I'm not super familiar with how out params are used in Lua, but presumably they still need to be passed into the function call. If they do, then it probably makes sense for them to be represented within params, but with a prefix or similar to indicate that they're out params. I can prefix these with `&` to represent an out param?

Reply:
Ah okay, I've spotted `apFunc->params[i]->unk28 & 0x200) != 0` in Type.cpp, which should be enough to go on. While having a look at this file, I noticed `Type::Execute()` doesn't seem to check `aArgs` has the correct length so it's possible some crashes are due to out of bounds array access. Should I file an issue for this?

Reply:
Type::Execute doesn't check because it depends on the number of out/optional params, it does check that there is enough arg to pop tho https://github.com/yamashi/CyberEngineTweaks/blob/master/src/reverse/Type.cpp#L142

Reply:
Ah yeah, my bad, I may not have had enough coffee when I was looking at that. Is there an example somewhere of what the Lua code when using out params anywhere?

Reply:
Alright, moved out params into return types in the signature